app/models/system.rb

```diff
@@ -1,6 +1,6 @@
 class System < ApplicationRecord
 
-  enum proxy_byos_mode: { payg: 0, byos: 1, hybrid: 2 }
+  enum proxy_byos_mode: { byos: 1, hybrid: 2 }
```

Comment:
The default value should be included in the enum (you were right the first time IMO).

Also, please include a validation limiting the field to enum keys, for example
```
validates :proxy_byos_mode, inclusion: { in: proxy_byos_mode.keys }
```

Reply:
Why now removing the `payg: 0` or shouldn't it be rather `payg: nil`?

Reply:
because when I ran tests on `payg: 0` all systems were marked as `proxy_byos_mode = 'payg'` that did not have `proxy_byos` set before which is not correct for RMT outside the public cloud scenario

Reply:
> The default value should be included in the enum (you were right the first time IMO).
> 
> Also, please include a validation limiting the field to enum keys, for example
> 
> ```
> validates :proxy_byos_mode, inclusion: { in: proxy_byos_mode.keys }
> ```

Validation is created by default, no need to set anything 
```
irb(main):007:0> System.create!(hostname: 'hostname', proxy_byos_mode: :aa)
Traceback (most recent call last):
        1: from (irb):7
ArgumentError ('aa' is not a valid proxy_byos_mode)
irb(main):008:0> 
```

Reply:
wrt to the default value and the enum,
yes, `enum proxy_byos_mode: { payg: 0, byos: 1, hybrid: 2 }` is more explicit
with that we would still the default to be `nil` to support non pubcloud scenario

which leads to the question _why would we have two ways to express the same thing_ 
I personally prefer 
```
# This value is only used in public cloud scenarios 
# It indicates that the system is using 
# PAYG (pay as you go) if nil, 
# BYOS (bring your own subscription) or 
# a mix of both (hybrid).
enum proxy_byos_mode: { byos: 1, hybrid: 2 }
```

Reply:
wrt to the default value and the enum,
yes, `enum proxy_byos_mode: { payg: 0, byos: 1, hybrid: 2 }` is more explicit
with that we would still need the default to be `nil` to support non pubcloud scenario

which leads to the question _why would we have two ways to express the same thing_ 
I personally prefer 
```
# This value is only used in public cloud scenarios 
# It indicates that the system is using 
# PAYG (pay as you go) if nil, 
# BYOS (bring your own subscription) or 
# a mix of both (hybrid).
enum proxy_byos_mode: { byos: 1, hybrid: 2 }
```

Reply:
wrt to the default value and the enum,
yes, `enum proxy_byos_mode: { payg: 0, byos: 1, hybrid: 2 }` is more explicit
with that we would still need the default to be `nil` to support non pubcloud scenario

which leads to the question _why would we have two ways to express the same thing_ 
I personally prefer 
```
# This value is only used in public cloud scenarios 
# It indicates that the system is using 
# PAYG (pay as you go) if nil, 
# BYOS (bring your own subscription) or 
# a mix of both (hybrid).
enum proxy_byos_mode: { byos: 1, hybrid: 2 }
```

Reply:
Thoughts between `nil, :payg, :byos, :hybrid` and `nil, :byos, :hybrid` ?